src/Server/Players/PlayersRepository.fs

```diff
@@ -44,20 +44,38 @@ module Database =
     }
 
   let getOrCreateByName connectionString (PlayerName name) : Task<Result<Player, exn>> =
-    task {
-      let! existingPlayer = getByName connectionString (PlayerName name)
-      match existingPlayer with
-      | Error ex -> return Error ex
-      | Ok (Some player) -> return Ok player
-      | Ok None ->
+    // Helper functions
+    let condExec f m = task {
+        match m with
+        | Error ex -> return Error ex
+        | Ok (Some player) -> return Ok (Some player)
+        | _ -> return! f()
+    }
+
+    let checkPlayerExists() = task {
+        let! existingPlayer = getByName connectionString (PlayerName name)
+        match existingPlayer with
+        | Error ex -> return Error ex
+        | Ok (Some player) -> return Ok (Some player)
+        | Ok None -> return Ok None
+    }
+
+    let tryToInsertPlayer() = task {
         let! affectedRows = insert connectionString name
         match affectedRows with
         | Error ex -> return Error ex
-        | Ok _ ->
-            let! player = getByName connectionString (PlayerName name)
-            match player with
-            | Ok (Some p) -> return Ok p
-            | Ok None -> return Error (exn "Player added but not found on a second look.")
-            | Error ex -> return Error ex
+        | Ok _ -> return Ok None
     }
+    // ----------------
+
+    // Logic main body
+    task {
+        let! myResult1 = checkPlayerExists()
+        let! myResult2 = condExec tryToInsertPlayer myResult1
+        let! myResult3 = condExec checkPlayerExists myResult2
```

Comment:
Do we need to name these `myResultn` or are there batter names for these?

Reply:
We can call them whatever you want... What would you like ?

Reply:
I suppose I'm asking more about what they all represent, and naming them accordingly.

Reply:
This is a difficult one, since they don't represent anything specific. Since the "result" cascades from one execution to the next.
for example, if `checkPlayerExists` returns a Player, result3 will hold that state that cascaded from the 1st execution. 
`tryToInsertPlayer` and the second `checkPlayerExists` will not execute, but the state will cascade into result3.

It would be nice to call it the same name and reassign the value, but immutability... hahaha

When we finally are doin the last match with result3, we could very well be checking state of result1

Since I have the language limitation, I don't know if I I explained it correctly.

Any ideas ?

Reply:
Just as an idea:
```fsharp
    task {
        let! checkPlayerResult = checkPlayerExists()
        let! insertPlayerResult = condExec tryToInsertPlayer checkPlayerResult
        let! finalResult = condExec checkPlayerExists insertPlayerResult

        match finalResult with
        | Error ex -> return Error ex
        | Ok (Some player) -> return Ok player
        | _ -> return Error (exn "Player added but not found on a second look.")
    }
```

What do you think ?

Reply:
I much prefer these names, but yes - it's difficult to name the intermediate results.